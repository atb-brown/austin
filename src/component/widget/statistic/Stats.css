.Stats {
  align-items: center;
  display: flex;
}

.Stats-Visual {
  background-color: rebeccapurple;
  border-radius: 10px;
  height: 100%;
}

.Stats-Container {
  flex-direction: row-reverse;
  height: 5vh;
  /* 96% is kind of a magic number that helps prevent drawing under a scroll bar. */
  width: 96%;
}
