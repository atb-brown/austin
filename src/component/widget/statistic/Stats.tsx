import HiddenStats from "./HiddenStats";
import "./Stats.css";
import WindowSize from "./WindowSize";
import React, { useCallback } from "react";

/**
 * A "ribbon" component that displays diagnostics/statistics.
 *
 * @return {ReactElement} React Component
 */
export default function Stats(): JSX.Element {
  const [showResults, setShowResults] = React.useState(true);
  const onClick = useCallback(() => {
    setShowResults(!showResults);
  }, [showResults, setShowResults]);

  if (showResults) {
    return (
      <div
        className="Stats Stats-Visual Stats-Container"
        data-testid="stats-ribbon"
        onClick={onClick}
      >
        <div
          className="Stats"
          style={{
            width: "10%",
          }}
        >
          <WindowSize />
        </div>
      </div>
    );
  } else {
    return <HiddenStats onClick={onClick} />;
  }
}
