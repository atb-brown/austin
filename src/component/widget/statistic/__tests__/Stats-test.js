import Stats from "../Stats";
import "@testing-library/jest-dom";
import { screen, render, fireEvent } from "@testing-library/react";

it("initial view renders to match snapshot", () => {
  const { asFragment } = render(<Stats />);

  expect(asFragment()).toMatchSnapshot();
});

it("hidden view renders to match snapshot", () => {
  const { asFragment } = render(<Stats />);

  const statsRibbon = screen.getByTestId("stats-ribbon");
  fireEvent.click(statsRibbon);

  expect(asFragment()).toMatchSnapshot();
});

it("click toggles the ribbon", () => {
  render(<Stats />);

  // Initial state, full ribbon is displayed.
  const statsRibbon = screen.getByTestId("stats-ribbon");
  expect(statsRibbon).toBeDefined();

  // After clicking, ribbon should be hidden.
  fireEvent.click(statsRibbon);
  const hiddenStatsRibbon = screen.getByTestId("stats-ribbon-hidden");
  expect(hiddenStatsRibbon).toBeDefined();
  expect(screen.queryByTestId("stats-ribbon")).toBeNull();

  // After clicking again, ribbon should be displayed again.
  fireEvent.click(hiddenStatsRibbon);
  expect(screen.getByTestId("stats-ribbon")).toBeDefined();
  expect(screen.queryByTestId("stats-ribbon-hidden")).toBeNull();
});
