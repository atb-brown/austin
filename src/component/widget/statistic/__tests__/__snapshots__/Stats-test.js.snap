// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hidden view renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    style="height: 5vh; display: flex; width: 96%; flex-direction: row-reverse; border-radius: 10px;"
  >
    <div
      data-testid="stats-ribbon-hidden"
      style="display: flex; height: 100%; align-items: center; flex-direction: row-reverse; background-color: rebeccapurple; justify-content: center; border-radius: 10px;"
    >
      â—€
    </div>
  </div>
</DocumentFragment>
`;

exports[`initial view renders to match snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="stats-ribbon"
    style="height: 5vh; display: flex; width: 97%; align-items: center; flex-direction: row-reverse; background-color: rebeccapurple; border-radius: 10px;"
  >
    <div
      style="display: flex; height: 100%; width: 10%; align-items: center;"
    >
      <div
        style="font-size: 10px; display: flex; flex-direction: column;"
      >
        <div>
          Window Size
        </div>
        <div>
          1024 x 768
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
